src/elasticodbc/qresult.c

```diff
@@ -308,12 +308,16 @@ void QR_add_message(QResultClass *self, const char *msg) {
         pos = 0;
         alsize = addlen + 1;
     }
-    if (message = realloc(message, alsize), NULL == message)
-        return;
-    if (pos > 0)
-        message[pos - 1] = ';';
-    strncpy_null(message + pos, msg, addlen + 1);
-    self->message = message;
+    char *message_tmp = realloc(message, alsize);
+    if (message_tmp) {
+        message = message_tmp;
+        if (pos > 0)
+            message[pos - 1] = ';';
+        strncpy_null(message + pos, msg, addlen + 1);
+        self->message = message;
+    } else {
+        free(message_tmp);
```

Comment:
Don't need to free, no memory was allocated.

Reply:
Cppcheck flagged this code, and this was the recommended fix: (memleakOnRealloc)
https://www.mantidproject.org/How_to_fix_CppCheck_Errors

Reply:
I think the initial code it flagged was due to message = realloc(message, .. ) which causes a memory leak on failure. 

This new code does not have that issue and the free won't be required. The memory leak detected should go away with just the if statement and message_tmp = realloc(message, ..)

Reply:
Removed else block